activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -327,12 +335,7 @@ def shard
       end
 
       def schema_cache
-        @pool.get_schema_cache(self)
-      end
-
-      def schema_cache=(cache)
-        cache.connection = self
-        @pool.set_schema_cache(cache)
+        @schema_cache ||= BoundSchemaReflection.new(@pool.schema_reflection, self)
```

Comment:
FWIW, I think this is [technically public API](https://edgeapi.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/AbstractAdapter.html#method-i-schema_cache-3D)

Reply:
Yeah, we even use it in our app.

Reply:
Hum, actually no, we use `Pool#set_schema_cache` which is private.